app/components/SearchableMetricSelector.js

```diff
@@ -1,25 +1,75 @@
 import { useEffect, useState } from 'react'
 
-function Modal(props){
-  const { supportedMetrics, onClose, selectMetric, show } = props;
-  const [searchText, setSearchText] = useState("")
-  if(!show){
-    return null;
+function sortDictionary(dictionary){
+    var sortedTagsArr = Object.entries(dictionary).sort((a,b) => b[1].length - a[1].length)
+    var retArr = [];
+    for(let i = 0; i < sortedTagsArr.length; i++){
+      var tempArr = sortedTagsArr[i]
+      var tempDict = {};
+      tempDict[tempArr[0]] = tempArr[1]
+      retArr.push(tempDict)
+    }
+    return retArr
+  }
+  
+//Create a dictionary of all unique tags/datasets that exist, each key holding an array of metrics belonging to that tag/dataset
+function createTagDictionary(supportedMetrics){
+  const groups = [...new Set(Object.keys(supportedMetrics).map(q => supportedMetrics[q].dataset))];
+  const metricRecords = Object.keys(supportedMetrics)
+  var tagDictionary = {};
+  
+  for (var i in groups){
+    var groupedRecords = [];
+    for(var j in Object.keys(supportedMetrics)){
+      var record = Object.keys(supportedMetrics)[j]
+      if(groups[i].indexOf(supportedMetrics[record].dataset) > -1 ){
```

Comment:
This should be `groups[i] === supportedMetrics[i]. dataset` since they should be exactly equal in order to belong to the group. You use the indexOf if you only want a partial match, such as in a search string.